### PR TITLE
test: added acceptance test to validate the e2e flow of the ERC Registry project

### LINKED ISSUE #1065: [ERC Registry] Phase 2: Add a suite of acceptance test to test the tool against localnode
### Problem  
The ERC Registry requires a suite of acceptance tests to ensure the tool functions correctly.

### Solution  
Implement a comprehensive suite of acceptance tests.

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:
This PR introduces a suite of acceptance tests to validate the end-to-end flow of the ERC registry project. These tests deploy a specified number of contracts, including ERC20, ERC721, and non-ERC contracts, to the target network defined in the environment. The suite then executes the ERC registry's main runner to index the entire target network, accurately detect ERC contracts, and correctly write them to the registry.

**Related issue(s)**:

Fixes #1065

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: tools/erc-repository-indexer/erc-contract-indexer/README.md [hunk 1]
```diff
@@ -35,12 +35,14 @@ The ERC Contract Indexer is a tool designed to facilitate the indexing and manag
...
+| `MIRROR_NODE_URL_WEB3` | The URL for the Hedera Mirror Node Web3Module API, required only when `HEDERA_NETWORK` is set to `local-node`. | Any value                                                                                                                                                                                                                       |
+| `STARTING_POINT`       | The starting point for contract indexing.                                                                      | A Hedera contract ID (e.g., `0.0.369`), an EVM 20-byte address (e.g., `0x0000000000000000000000000000000000000369`), or a get contract list next pointer (e.g., `/api/v1/contracts?limit=100&order=asc&contract.id=gt:0.0.369`) |
+| `SDK_OPERATOR_ID`      | The operator ID used to initialize a Hashgraph SDK, applicable only for testing.                               | Hedera Account IDs                                                                                                                                                                                                              |
```
Comment (Nana-EC) [inline tools/erc-repository-indexer/erc-contract-indexer/README.md:44]: If just for testing pull them out of this section and either put in a testing section or at least a separate table for testing

File: tools/erc-repository-indexer/erc-contract-indexer/tests/acceptance/acceptance.test.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import testHelper from './utils/helper';
+import testConstants from './utils/constants';
+import { ercRegistryRunner } from '../../src/runner';
+import { Helper } from '../../src/utils/helper';
+import constants from '../../src/utils/constants';
+import NodeClient from '@hashgraph/sdk/lib/client/NodeClient';
+
+describe('ERC Registry Acceptance Test', () => {
+  // production networks take more time to finish deployments
+  jest.setTimeout(60000);
```
Comment (Nana-EC) [inline tools/erc-repository-indexer/erc-contract-indexer/tests/acceptance/acceptance.test.ts:31]: nit: i thought there used to be a way to test/poll that local node was ready. 
@natanasow are you aware of any reliable way?
Nothing to do here but something to think about

**Review by Nana-EC — APPROVED** [review]
LG, minor comments

File: tools/erc-repository-indexer/erc-contract-indexer/README.md (hunk 1 above)
Comment (quiet-node) [reply tools/erc-repository-indexer/erc-contract-indexer/README.md:44]: Gotcha will update in follow up PR!

File: tools/erc-repository-indexer/erc-contract-indexer/tests/acceptance/acceptance.test.ts (hunk 2 above)
Comment (quiet-node) [reply tools/erc-repository-indexer/erc-contract-indexer/tests/acceptance/acceptance.test.ts:31]: Gotcha thanks for bringing it up!

**Pull request merged**